added microphone examples for node and browser

mreinstein opened the pull request:
shows minimal code to take microphone input and render it with `waveform` :)

mreinstein committed:
added microphone examples for node and browser

mreinstein commented:
@dfcreative any thoughts on these?

dy commented:
@mreinstein thank you for the contribution, I will try to include these examples with updated renderer (it takes _gl-waveform_, which takes _signal-tree_ to be done).

dy commented on package.json:
```diff
@@ -30,7 +31,9 @@
     "audio-speaker": "^1.1.2",
-    "watchify": "^3.6.0"
+    "mic": "^2.1.1",
+    "watchify": "^3.6.0",
```
gotta switch to _budo_, no need in watchify anymore

mreinstein replied:
we could use *budo*, but it's a lot more complex than watchify. And the example works by opening the `test.html` directly from a folder. No need to have a server to run it.

mreinstein replied:
@dfcreative any thoughts on this?

dy replied:
@mreinstein why budo is complex? I mean we should not add it to dependency of course, but `budo test` for me is pretty simple :smile: 
Also there is CORS-limitations on running files/mic from locally opened file.

mreinstein replied:
@dfcreative 

> why budo is complex?

`watchify` is _dead_ simple. It just watches the file for changes and runs browserify on it. `budo` has a bit more of the kitchen sink, (css watching, plugin architecture, etc.)

> but `budo test` for me is pretty simple

It's definitely simple to run, but has a lot of stuff going on under the hood. I guess it's just personal tastes. I like minimal things (even dev tools) that have less features but are smaller and specific.

That said, if you really like `budo` I don't want to force anyone into my preferences. :) I'm happy to update this to budo if you really want it. 👍

mreinstein replied:
> there is CORS-limitations on running files/mic from locally opened file

in Chrome at least, I'm able to click the `examples/microphone-browser/index.html` file, click "allow" from the "use microphone" popup dialog, and see the waveform render.

dfcreative commented on examples/microphone-node/index.js:
```diff
@@ -0,0 +1,27 @@
+'use strict'
+
+const waveform = require('../../')
+const mic      = require('mic')
```
Just wonder does it work in node :smile:? I never tested it

mreinstein committed:
updated examples to use budo

mreinstein commented:
@dfcreative I see what you mean about `budo`. Definitely gets rid of some of the watchify/serving boilerplate. I've updated the PR to use it, and removed the old plumbing. Is the PR good to go?

mreinstein commented:
@dfcreative not to be a pest but any chance of getting this merged?
